### ISSUE TITLE
[BUG] SKBitmap.Encode issue with Skiasharp 2.88.6 on .NET 8.0 MacOS and iOS

### ISSUE THREAD

**Issue by chinnumuniyappan** [mono/SkiaSharp#2666]
### Description

When encoding images with SkiaSharp 2.88.6 on. NET 8. 0 for MacOS and iOS, there are preservation issues with the resulting images. However, in the previous version of SkiaSharp, the encoding was done correctly. The same image is encoded properly on Windows and Andriod.

Skiasharp version: 2.88.6
Platform: MacIO and iOS.

### Code

        SKBitmap bitmap = SKBitmap.Decode(imageStream);

        MemoryStream outputImage = new MemoryStream();
        bitmap.Encode(SKEncodedImageFormat.Jpeg, 100).SaveTo(outputImage);

### Version of SkiaSharp

Other (Please indicate in the description)

### Last Known Good Version of SkiaSharp

2.88.2 (Previous)

### IDE / Editor

Visual Studio Code (macOS)

### Platform / Operating System

macOS

### Platform / Operating System Version

MacOS 13.2
iOS 16.4

### Devices

-

### Relevant Screenshots

Original image: 

![pdf_succinctly_page3](https://github.com/mono/SkiaSharp/assets/65020530/614333ca-f924-40fb-9cb6-33c013a1e8f1)

Encoded using Skasharp 2.88.6

![output-encoded](https://github.com/mono/SkiaSharp/assets/65020530/2ba85a3e-5798-4952-8145-9df96bae7c63)

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

**Comment by KRA2008** [comment]
likely duplicate of #2650

**Comment by gunasekarant** [comment]
Any expected date for this bug fix?

**Comment by jwalaszek** [comment]
I also see this issue using SkiaSharp 2.88.6 on macOS .NET 8. After downgrading to 2.88.5 the issue is gone.

**Comment by mattleibow** [comment]
I tested 3.0.1 and the latest main and I think this is an issue in libjpeg-turbo. For now I am reverting to the latest 2.x series.

I had a look at the original security issue and it does not affect the 2.x series because 12-bit jpegs are not actually supported in our build. The 2.x series required the entire library to be either 8-bit or 12-bit, and we use 8-bit. The 3.x series allowed for both simultaneously.

The security issue was just reported for 2.1.90 (which seems like a 2.x series but is actually a 3.0 beta). The current stable version is 2.1.5.1 which is on a separate branch.

I have opened an issue in libjpeg-turbo where I will try and determine the real cause and where this issue should live: https://github.com/libjpeg-turbo/libjpeg-turbo/issues/741
